src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -0,0 +1,391 @@
+                            //If the predecessor doesn't exist, then update the Data import object with an error and continue with the next dataImport.
+                            if (!predecessorExists(objWrap,predecessor)) {
+                                dataImport.put(UTIL_Namespace.adjustNPSPNamespace(objMapping.Imported_Record_Status_Field_Name__c),System.label.bdiAdditionalObjPredNotFound);
+                                continue;
+                            }
+
+                        } else { 
+
+                            //If there is an existing record then we don't need to perform the normal validation and can proceed to populating
+                            //the record with any non-null values.
+                            objWrap.existingSobjectId = (Id)dataImport.get(UTIL_Namespace.adjustNPSPNamespace(objMapping.Imported_Record_Field_Name__c));
+
+                        }
+
+                        objWrapsForProcessing.add(objWrap);
+                    }
+
+                    if (!dataImportService.isDryRun && objWrapsForProcessing.size() > 0) {
+                        objWrapsForProcessing = objMappingLogic.populateObjects(objWrapsForProcessing);
+
+                        for (BDI_ObjectWrapper objWrap : objWrapsForProcessing) {
+
+                            // If the sObject in the objWrap is not null then add it to the map of object wraps by sobject type to upsert.
+                            // They are grouped by object api name so that we only need to do one upsert per object per tier.
+                            if (objWrap.sObj != null) {
+                                if (objAPINameToObjWraps.get(objMapping.Object_API_Name__c) != null) {
+                                    objAPINameToObjWraps.get(objMapping.Object_API_Name__c).add(objWrap);
+                                } else {
+                                    objAPINameToObjWraps.put(objMapping.Object_API_Name__c,new BDI_ObjectWrapper[]{objWrap});
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return objAPINameToObjWraps;
+    }
+
+    /*******************************************************************************************************
+    * @description Takes a map of Sobject api name to the object wrappers for that sobject and either updates or inserts the data.
+    * @param objAPINameToObjWraps map of Sobject api name to object wrappers.
+    */
+
+    private void importObjectWrappers (Map<String,BDI_ObjectWrapper[]> objAPINameToObjWraps) {
+        //If there are any records to create, then combine them into lists for update
+        if (objAPINameToObjWraps.size() > 0) {
+            for (String objApiName : objAPINameToObjWraps.keySet()) {
+
+                BDI_ObjectWrapper[] objWraps = objAPINameToObjWraps.get(objApiName);
+                
+                Sobject[] sObjsForUpdate = new Sobject[]{};
```

Comment:
Style guide suggest "SObject" over "Sobject"

Reply:
This was updated.